src/Lecture3.hs

```diff
@@ -34,7 +34,7 @@ module Lecture3
     , apply
     ) where
 
-
+import Data.Semigroup
```

Comment:
I see GHC warning that this import is redundant ⚠️ 
Probably you can remove it 🧹